internal/e2e/basic_test.go

```diff
@@ -24,5 +27,11 @@ steps:
 		t.Errorf("Build state = %q, want %q", got, want)
 	}
 
-	// TODO: add ability to inspect job logs
+	logs, err := tc.fetchLogs(ctx, build)
+	if err != nil {
+		t.Errorf("tc.fetchLogs() error = %v", err)
+	}
```

Comment:
this error handling and the waitForBuild error handling can all happen in the callee side I think, I don't think they will ever be useful on caller side? 

Since we have a `t *testing.T` available, we can effectively "throw" error from callee side.

Reply:
That's probably true. I'm not sure we'll need a case where they can fail, so I'll move that.